pkg/event/target/mqtt.go

```diff
@@ -159,7 +133,7 @@ func (target *MQTTTarget) Close() error {
 }
 
 // NewMQTTTarget - creates new MQTT target.
-func NewMQTTTarget(id string, args MQTTArgs) (*MQTTTarget, error) {
+func NewMQTTTarget(id string, args MQTTArgs, doneCh chan struct{}) (*MQTTTarget, error) {
```

Comment:
Accept read channel when read is primary.  When a  function creates a channel, any writer function should close the channel. Similarly all reader should not close channel.  Hence accept channels specifically i.e. channel argument should be a read channel or writer channel.

Reply:
modified the argument :+1: